src/classes/ADDR_Addresses_TEST.cls

```diff
@@ -998,6 +998,254 @@ public with sharing class ADDR_Addresses_TEST {
+        for(Address__c eachAddress : listAddr) {
+            if(eachAddress.Default_Address__c) {
+              accountIdToAddress.put(eachAddress.Household_Account__c, eachAddress);
+            }
+        }
+
+        // Verify the Account's Billing Address info matches the Default Address record
+        List<Account> retrievedAccounts = Database.query(buildAccountSoql());
+        assertEquals(listAddrT[0], accountIdToAddress.get(retrievedAccounts[0].Id));
+
+        // Verify the Contact's Address was updated as expected and that the Household and Contact Addresses are different.
+        Map<Id, Account> accountIdToAccount = new Map<Id, Account>(retrievedAccounts);
+        List<Contact> listCon = Database.query(buildContactSoql());
+        for (Contact eachContact : listCon) {
+            System.assertEquals(true, eachContact.MailingState.equals('California'));
+            System.assertEquals(true, eachContact.is_Address_Override__c);
+            System.assertNotEquals(null, eachContact.Current_Address__c);
+
+            Account eachContactAccount = accountIdToAccount.get(eachContact.AccountId);
+            System.assertEquals(false, isMatchAddressAccCon(eachContactAccount, eachContact));
+        }
+
+        // Verify the New Address created matches the Contact's Address and not set the Default
+        for (Integer i = 0; i < listAddr.size(); i++) {
+            if(listAddr[i].MailingState__c.equals('California')) {
+                assertEqualAddresses(listAddr[i], listCon[i]);
+                System.assertEquals(false, listAddr[i].Default_Address__c);
+            }
+        }
+    }
+
+
+    /*********************************************************************************************************
+    @description
+        Update Address' Mailing Country and State (with State & Country Picklists enabled) when the Country is empty
+        or the State empty and/or mismatched.
+    verify:
+        New Address' Country and State fields are updated with the source sObject's CountryCode and StateCode values
+    **********************************************************************************************************/
+    static testMethod void testUpdateAddressCountryAndStateWithAccountOrContactCountryCodeAndStateCode() {
+        if (!ADDR_Addresses_TDTM.isStateCountryPicklistsEnabled) {
+            return;
+        }
+
+        if (strTestOnly != '*' && strTestOnly != 'testUpdateAddressCountryAndStateWithAccountOrContactCountryCodeAndStateCode') {
+            return;
+        }
+
+        Integer hhSize = 1;
+        Integer contactSize = 1;
+
+        // This creates a default Address for each HH
+        createHHTestData(hhSize, contactSize);
+
+        Contact con = listConT[0];
+        con.put('MailingStateCode', 'CA');
+        con.put('MailingCountryCode', 'US');
+        con.is_Address_Override__c = true;
+
+        Address__c addressWithoutCountryState = listAddrT[0];
```

Comment:
Suggestion: since test method name is clear what is tested and expected, and since the scope of the variable is local (small scope), we might just use `address` as the name?